crates/consensus-logic/src/csm/worker.rs

```diff
@@ -376,12 +376,22 @@ fn apply_action(
 
         // Update checkpoint entry in database to mark it as included in L1.
         SyncAction::UpdateCheckpointInclusion {
-            epoch,
+            checkpoint,
             l1_reference,
         } => {
-            let Some(mut ckpt_entry) = ckpt_db.get_checkpoint_blocking(epoch)? else {
-                warn!(%epoch, "missing checkpoint we wanted to set the state of, ignoring");
-                return Ok(());
+            let epoch = checkpoint.batch_info().epoch();
+
+            let mut ckpt_entry = match ckpt_db.get_checkpoint_blocking(epoch)? {
+                Some(c) => c,
+                None => {
+                    warn!(%epoch, "missing checkpoint we wanted to set the state of, creating new entry");
+
+                    CheckpointEntry::new(
+                        checkpoint,
+                        CheckpointProvingStatus::ProofReady,
+                        CheckpointConfStatus::Pending,
+                    )
+                }
             };
 
             ckpt_entry.confirmation_status = CheckpointConfStatus::Confirmed(l1_reference);
```

Comment:
When we're constructing a new one, can't we preemptively set it as `Confirmed` here?  I suppose it doesn't really matter.

Reply:
Yeah, it doesn't matter. I just didn't want to re-set the same status twice.